orderbook/src/fee.rs

```diff
@@ -213,6 +229,17 @@ impl MinFeeCalculator {
         let price = estimate.price_in_sell_token_f64(&query);
         Ok(U256::from_f64_lossy(fee_in_eth * price))
     }
+
+    fn calculate_fee_factor(&self, app_data: Option<AppId>) -> f64 {
+        app_data
+            .and_then(|app_data| self.partner_additional_fee_factors.get(&app_data).cloned())
+            .unwrap_or(1.0)
+            * self.fee_factor
+    }
+
+    fn apply_fee_factor(fee: U256, factor: f64) -> U256 {
```

Comment:
Just curious, why do you prefer this as a associated function vs a "free" function in the module?

Reply:
Generally, my preference for associated functions is a habit from C++. In this case though, I thought it'll make more sense to keep all functions related to the same logic in one place.

Reply:
> I thought it'll make more sense to keep all functions related to the same logic in one place.

Interesting. In my mind, I tend to like to use modules to split up logic into groups instead of types (i.e. associated functions). Not suggesting one is better than the other, just interested in your PoV :smile:.